### PR TITLE
Fix to the overflowing of tables off the page

### PR DESCRIPTION
**Pull request by Louis-Boulet-HC**
- added block that truncates dataframe based on df width in terms of characters until it is expected to fit on the page
- takes into account font size and page orientation (landscape vs portrait)

TO DO:
- get pylatex to apply the fontsize onto the pdf document output of the table

### CODE REVIEW

**Review by fnwaohaHC — APPROVED** [review]

**Comment by Louis-Boulet-HC** [comment]
To note: 3 test cases currently fail but they have already been fixed in separate branches and merged into main

**Review by zgaffarHC — APPROVED** [review]

File: table_processing/GeneratedTable.py [hunk 1]
```diff
@@ -5,41 +5,47 @@
...
+        self.font_size = font_size
         self.geometry_options = {
             'margin':margin,
-            'landscape':True
+            'landscape':landscape
         }
         self.multi_row = multi_row
-        self.row_height = row_height
+        if row_height < 1.25:
+            self.row_height = 1.25
```
Comment (pgaviganHC) [inline table_processing/GeneratedTable.py:27]: I'm a bit wary of magic numbers. Can these be changed to meaningfully named constants?

File: table_processing/GeneratedTable.py [hunk 2]
```diff
@@ -5,41 +5,47 @@
...
+        else:
+            self.row_height = row_height  # row height value is relative to font size (so automatically adjusts with it)
+
+        if self.multi_row == True and (self.row_lines == False or self.vertical_lines == False):
+            raise Exception('Cannot create table with multi-rows and with no vertical and horizontal lines')
+        
+        if self.columns > 15:
+            self.columns = 15
+            logging.warning('Cannot generate that many columns, shrunk down to 15')
```
Comment (pgaviganHC) [inline table_processing/GeneratedTable.py:36]: A few more magic numbers to check on

**Review by pgaviganHC — CHANGES_REQUESTED** [review]
There are three failing tests from this PR. Please investigate and correct the issues.

**Pull request merged**